### PR TITLE
Domains: Show "Free with your plans" for incoming transfer products

### PR DESCRIPTION
**Pull request by hambai**
Incoming transfers are also free with a plan, so this PR should display the "Free with your plan" notice next to the Domain Transfer and Domain Transfer Privacy products in the cart.

before:
screen shot 2017-11-22 at 15 20 13

after:
screen shot 2017-11-22 at 22 54 31

Also I've fixed the Privacy Protection card on the checkout screen:
screen shot 2017-11-22 at 22 55 00

### CODE REVIEW

File: client/lib/products-values/index.js [hunk 1]
```diff
@@ -249,6 +253,13 @@ export function isDomainTransfer( product ) {
 	return isTransfer( product );
 }
 
+export function isDomainTransferPrivacy( product ) {
+	product = formatProduct( product );
+	assertValidProduct( product );
+
+	return product.product_slug === 'domain_transfer_privacy';
```
Comment (klimeryk) [inline client/lib/products-values/index.js]: This should be the `TRANSFER_IN_PRIVACY` constant from `lib/domains/constants.js`.

File: client/lib/products-values/index.js [hunk 2]
```diff
@@ -210,7 +210,11 @@ export function isDomainProduct( product ) {
 	assertValidProduct( product );
 
 	return (
-		isDomainMapping( product ) || isDomainRegistration( product ) || isPrivacyProtection( product )
+		isDomainMapping( product ) ||
+		isDomainRegistration( product ) ||
+		isPrivacyProtection( product ) ||
+		isDomainTransfer( product ) ||
+		isDomainTransferPrivacy( product )
```
Comment (klimeryk) [inline client/lib/products-values/index.js]: Seems like we don't need the check [here](https://github.com/Automattic/wp-calypso/blob/dc3426b6bf30be7eb62734b6d92d043bb3107bcd/client/me/purchases/purchase-item/index.jsx#L154) if transfers in are a domain product.

Comment (aidvu) [reply client/lib/products-values/index.js]: Not the same atm: https://github.com/Automattic/wp-calypso/blob/5f15f3365213253724ca0d8798c6cb8160f00bc9/client/lib/products-values/index.js#L208

We can review and see if it makes sense to set it as a domain product.

**Comment by hambai** [comment]
OK, so I removed the domain transfer products from the `isDomainProduct` check - it makes sense to treat those as separate products. I fixed my code accordingly. I also took the liberty to refactor and remove the `isTransfer` function as it seems something that we should check in `product-values`.

**Comment by aidvu** [comment]
Hmmm... Nevermind my comment. Somehow I didn't see the free-with-plan. Works ok now.

**Pull request merged**